### ISSUE TITLE
IDataManager should not specify a transaction_manager attribute

### ISSUE THREAD

**Issue by jimfulton** [zopefoundation/transaction#47]
It's not used by the transaction package and none of its concern.

**Comment by jimfulton** [comment]
I noticed this while implementing a data manager for which this attribute makes no sense.

**Comment by jenstroeger** [comment]
Agreed, I’ve deleted the [tm](https://transaction.readthedocs.io/en/latest/api.html#transaction.interfaces.ITransactionManager) reference from two [IDataManager](https://transaction.readthedocs.io/en/latest/api.html#transaction.interfaces.IDataManager) implementations. Other than a code-smell, I can’t think of a scenario where a data manager would need a back-reference to the owning tm.

**Comment by d-maurer** [comment]
@jimfulton 
> I noticed this while implementing a data manager for which this attribute makes no sense.

How does your "data manager" determines the "current transaction" to potentially jjoin it?
